pac4j-core/src/main/java/org/pac4j/core/client/IndirectClient.java

```diff
@@ -40,7 +40,7 @@ protected void internalInit(final WebContext context) {
     }
 
     @Override
-    public final HttpAction redirect(final WebContext context) throws HttpAction {
+    public HttpAction redirect(final WebContext context) throws HttpAction {
```

Comment:
The CallbackUrlResolver is called one time during the construction and the value computed is stored in OAuthConfig which is a properties of the OAuthService, I don't know how I could make it work with a custom CallBackUrlResolver.